File: gesttravaux-symfony/src/Entity/Inspecteur.php
<?php

namespace App\Entity;

use App\Repository\InspecteurRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

/**
 * Inspecteur entity - Site inspectors
 *
 * Represents inspectors who monitor construction sites,
 * add observations, and upload documents
 *
 * @author GestTravaux Pro
 */
#[ORM\Entity(repositoryClass: InspecteurRepository::class)]
#[ORM\Table(name: 'inspecteurs')]
class Inspecteur
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'idInspecteur')]
    private ?int $id = null;

    #[ORM\Column(length: 45)]
    private ?string $nomInspecteur = null;

    #[ORM\Column(length: 45)]
    private ?string $prenomInspecteur = null;

    #[ORM\Column(length: 45)]
    private ?string $emailInspecteur = null;

    #[ORM\Column(length: 45)]
    private ?string $telInspecteur = null;

    #[ORM\Column(length: 45)]
    private ?string $secteurInspecteur = null;

    #[ORM\OneToOne(targetEntity: User::class, inversedBy: 'inspecteur')]
    #[ORM\JoinColumn(nullable: true)]
    private ?User $user = null;

    /**
     * @var Collection<int, Chantier>
     */
    #[ORM\OneToMany(targetEntity: Chantier::class, mappedBy: 'inspecteur')]
    private Collection $chantiers;

    public function __construct()
    {
        $this->chantiers = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomInspecteur(): ?string
    {
        return $this->nomInspecteur;
    }

    public function setNomInspecteur(string $nomInspecteur): static
    {
        $this->nomInspecteur = $nomInspecteur;

        return $this;
    }

    public function getPrenomInspecteur(): ?string
    {
        return $this->prenomInspecteur;
    }

    public function setPrenomInspecteur(string $prenomInspecteur): static
    {
        $this->prenomInspecteur = $prenomInspecteur;

        return $this;
    }

    public function getEmailInspecteur(): ?string
    {
        return $this->emailInspecteur;
    }

    public function setEmailInspecteur(string $emailInspecteur): static
    {
        $this->emailInspecteur = $emailInspecteur;

        return $this;
    }

    public function getTelInspecteur(): ?string
    {
        return $this->telInspecteur;
    }

    public function setTelInspecteur(string $telInspecteur): static
    {
        $this->telInspecteur = $telInspecteur;

        return $this;
    }

    public function getSecteurInspecteur(): ?string
    {
        return $this->secteurInspecteur;
    }

    public function setSecteurInspecteur(string $secteurInspecteur): static
    {
        $this->secteurInspecteur = $secteurInspecteur;

        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): static
    {
        $this->user = $user;

        return $this;
    }

    /**
     * @return Collection<int, Chantier>
     */
    public function getChantiers(): Collection
    {
        return $this->chantiers;
    }

    public function addChantier(Chantier $chantier): static
    {
        if (!$this->chantiers->contains($chantier)) {
            $this->chantiers->add($chantier);
            $chantier->setInspecteur($this);
        }

        return $this;
    }

    public function removeChantier(Chantier $chantier): static
    {
        if ($this->chantiers->removeElement($chantier)) {
            // set the owning side to null (unless already changed)
            if ($chantier->getInspecteur() === $this) {
                $chantier->setInspecteur(null);
            }
        }

        return $this;
    }

    /**
     * Get full name for display
     */
    public function getFullName(): string
    {
        return $this->prenomInspecteur . ' ' . $this->nomInspecteur;
    }

    public function __toString(): string
    {
        return $this->getFullName();
    }
}

File: gesttravaux-java/src/main/java/com/immosync/gesttravaux/models/Inspecteur.java
package com.immosync.gesttravaux.models;

import jakarta.persistence.*;
import java.util.HashSet;
import java.util.Set;

/**
 * Entity representing an inspector who supervises construction sites.
 * Inspectors can be assigned to multiple chantiers.
 *
 * @author GestTravaux Pro
 * @version 1.0
 */
@Entity
@Table(name = "INSPECTEURS")
public class Inspecteur {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idInspecteur")
    private Integer idInspecteur;

    @Column(name = "nomInspecteur", length = 45, nullable = false)
    private String nomInspecteur;

    @Column(name = "prenomInspecteur", length = 45, nullable = false)
    private String prenomInspecteur;

    @Column(name = "emailInspecteur", length = 45, nullable = false)
    private String emailInspecteur;

    @Column(name = "telInspecteur", length = 45)
    private String telInspecteur;

    @Column(name = "secteurInspecteur", length = 45)
    private String secteurInspecteur;

    @OneToMany(mappedBy = "inspecteur", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private Set<Chantier> chantiers = new HashSet<>();

    /**
     * Default constructor required by JPA.
     */
    public Inspecteur() {
    }

    /**
     * Constructor with essential fields.
     *
     * @param nomInspecteur inspector last name
     * @param prenomInspecteur inspector first name
     * @param emailInspecteur inspector email
     */
    public Inspecteur(String nomInspecteur, String prenomInspecteur, String emailInspecteur) {
        this.nomInspecteur = nomInspecteur;
        this.prenomInspecteur = prenomInspecteur;
        this.emailInspecteur = emailInspecteur;
    }

    // Getters and Setters

    /**
     * Gets the inspector ID.
     *
     * @return the inspector ID
     */
    public Integer getIdInspecteur() {
        return idInspecteur;
    }

    /**
     * Sets the inspector ID.
     *
     * @param idInspecteur the inspector ID
     */
    public void setIdInspecteur(Integer idInspecteur) {
        this.idInspecteur = idInspecteur;
    }

    /**
     * Gets the inspector last name.
     *
     * @return the last name
     */
    public String getNomInspecteur() {
        return nomInspecteur;
    }

    /**
     * Sets the inspector last name.
     *
     * @param nomInspecteur the last name
     */
    public void setNomInspecteur(String nomInspecteur) {
        this.nomInspecteur = nomInspecteur;
    }

    /**
     * Gets the inspector first name.
     *
     * @return the first name
     */
    public String getPrenomInspecteur() {
        return prenomInspecteur;
    }

    /**
     * Sets the inspector first name.
     *
     * @param prenomInspecteur the first name
     */
    public void setPrenomInspecteur(String prenomInspecteur) {
        this.prenomInspecteur = prenomInspecteur;
    }

    /**
     * Gets the inspector email.
     *
     * @return the email
     */
    public String getEmailInspecteur() {
        return emailInspecteur;
    }

    /**
     * Sets the inspector email.
     *
     * @param emailInspecteur the email
     */
    public void setEmailInspecteur(String emailInspecteur) {
        this.emailInspecteur = emailInspecteur;
    }

    /**
     * Gets the inspector phone number.
     *
     * @return the phone number
     */
    public String getTelInspecteur() {
        return telInspecteur;
    }

    /**
     * Sets the inspector phone number.
     *
     * @param telInspecteur the phone number
     */
    public void setTelInspecteur(String telInspecteur) {
        this.telInspecteur = telInspecteur;
    }

    /**
     * Gets the inspector sector/area.
     *
     * @return the sector
     */
    public String getSecteurInspecteur() {
        return secteurInspecteur;
    }

    /**
     * Sets the inspector sector/area.
     *
     * @param secteurInspecteur the sector
     */
    public void setSecteurInspecteur(String secteurInspecteur) {
        this.secteurInspecteur = secteurInspecteur;
    }

    /**
     * Gets the chantiers assigned to this inspector.
     *
     * @return set of chantiers
     */
    public Set<Chantier> getChantiers() {
        return chantiers;
    }

    /**
     * Sets the chantiers assigned to this inspector.
     *
     * @param chantiers set of chantiers
     */
    public void setChantiers(Set<Chantier> chantiers) {
        this.chantiers = chantiers;
    }

    /**
     * Gets the full name of the inspector.
     *
     * @return full name (first name + last name)
     */
    public String getFullName() {
        return prenomInspecteur + " " + nomInspecteur;
    }

    @Override
    public String toString() {
        return getFullName();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Inspecteur)) return false;
        Inspecteur that = (Inspecteur) o;
        return idInspecteur != null && idInspecteur.equals(that.getIdInspecteur());
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}
